### PR TITLE
feat: Add more AI options

### PR DESCRIPTION
**Pull request by muhsin-k**
https://github.com/chatwoot/chatwoot/assets/12408980/d0b33211-ff43-4aef-b0f8-4f974a436b95

### CODE REVIEW

File: app/javascript/dashboard/i18n/locale/en/integrations.json [hunk 1]
```diff
@@ -83,6 +83,28 @@
...
+        "MAKE_FRIENDLY": "Make Friendly",
+        "MAKE_FORMAL": "Make Formal",
+        "SIMPLIFY": "Simplify"
+      },
+      "ASSISTANCE_MODAL": {
+        "DRAFT_TITLE": "Draft content",
+        "GENERATED_TITLE": "Generated content",
+        "AI_WRITING": "AI is writing",
+        "BUTTONS": {
+          "APPLY": "Apply",
```
Comment (nithindavid) [inline app/javascript/dashboard/i18n/locale/en/integrations.json:104]: "Use this suggestion" or "Insert into editor"? 

Toughts @scmmishra @Hricha-Shandily

File: app/javascript/dashboard/components/widgets/AIAssistanceModal.vue [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        {{ $t('INTEGRATION_SETTINGS.OPEN_AI.ASSISTANCE_MODAL.DRAFT_TITLE') }}
+      </h4>
+      <p v-if="draftMessage">
+        {{ draftMessage }}
+      </p>
+      <h4 v-if="draftMessage" class="sub-block-title margin-top-1  w-full">
+        {{
+          $t('INTEGRATION_SETTINGS.OPEN_AI.ASSISTANCE_MODAL.GENERATED_TITLE')
+        }}
+      </h4>
```
Comment (nithindavid) [inline app/javascript/dashboard/components/widgets/AIAssistanceModal.vue:15]:
```suggestion
<div v-if="draftMessage">
      <h4 class="sub-block-title margin-top-1 w-full">
        {{ $t('INTEGRATION_SETTINGS.OPEN_AI.ASSISTANCE_MODAL.DRAFT_TITLE') }}
      </h4>
      <p>
        {{ draftMessage }}
      </p>
      <h4 class="sub-block-title margin-top-1  w-full">
        {{
          $t('INTEGRATION_SETTINGS.OPEN_AI.ASSISTANCE_MODAL.GENERATED_TITLE')
        }}
      </h4>
</div>
```

**Review by nithindavid — DISMISSED** [review]

**Comment by nithindavid** [comment]
@muhsin-k Can we use 
image
 for Fix grammar 
 
 For expand 
image

For shortening
image

cc @scmmishra

File: app/javascript/dashboard/i18n/locale/en/integrations.json (hunk 1 above)
Comment (scmmishra) [reply app/javascript/dashboard/i18n/locale/en/integrations.json:104]: "Use this suggestion" sounds great

**Review by scmmishra — CHANGES_REQUESTED** [review]
CleanShot 2023-07-14 at 12 18 39@2x
The layout seems off

---

CleanShot 2023-07-14 at 12 19 44@2x

When I shorten the content, the display looks like this

File: app/javascript/dashboard/i18n/locale/en/integrations.json [hunk 3]
```diff
@@ -83,6 +83,28 @@
       "CREATE_ERROR": "There was an error creating a meeting link, please try again"
     },
     "OPEN_AI": {
+      "AI_ASSIST": "AI Assist",
+      "WITH_AI": "With AI",
```
Comment (scmmishra) [inline app/javascript/dashboard/i18n/locale/en/integrations.json:87]: Using sentence case here, also maybe we should use interpolation here if possible, some languages may have the operator and noun before the subject.

**Review by scmmishra — CHANGES_REQUESTED** [review]
CleanShot 2023-07-14 at 12 27 10@2x

When the editor is empty, is it possible to show the rest of the options and keep it disabled, right now when I click that button it feels that there's only one option and I may not open it again

File: app/javascript/dashboard/components/widgets/AIAssistanceModal.vue [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        });
+        const {
+          data: { message: generatedMessage },
+        } = result;
+        this.generatedContent = generatedMessage;
+      } catch (error) {
+        this.showAlert(this.$t('INTEGRATION_SETTINGS.OPEN_AI.GENERATE_ERROR'));
+      } finally {
+        this.isGenerating = false;
+      }
```
Comment (scmmishra) [inline app/javascript/dashboard/components/widgets/AIAssistanceModal.vue:101]: We can move this to the `aIMixin`

File: app/javascript/dashboard/components/widgets/AIAssistanceModal.vue [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+          conversationId: this.conversationId,
+        });
+        const {
+          data: { message: generatedMessage },
+        } = result;
+        this.generatedContent = generatedMessage;
+      } catch (error) {
+        this.showAlert(this.$t('INTEGRATION_SETTINGS.OPEN_AI.GENERATE_ERROR'));
+      } finally {
+        this.isGenerating = false;
```
Comment (scmmishra) [inline app/javascript/dashboard/components/widgets/AIAssistanceModal.vue:100]: I see the default value is true in the data, let's also set the value to true explicitly at the start of the method

**Review by scmmishra — CHANGES_REQUESTED** [review]

**Comment by muhsin-k** [comment]
> CleanShot 2023-07-14 at 12 27 10@2x
> When the editor is empty, is it possible to show the rest of the options and keep it disabled, right now when I click that button it feels that there's only one option and I may not open it again

@scmmishra I had the same thought, but unfortunately, the ninja-keys library does not support displaying disabled items. I will investigate whether it is possible to fork the library and add this feature myself. All the other review comments have been addressed.

**Review by scmmishra — APPROVED** [review]

File: app/javascript/dashboard/i18n/locale/en/integrations.json [hunk 6]
```diff
@@ -83,6 +83,28 @@
...
+      "AI_ASSIST": "AI Assist",
+      "WITH_AI": " %{option} with AI ",
+      "OPTIONS": {
+        "REPLY_SUGGESTION": "Reply Suggestion",
+        "SUMMARIZE": "Summarize",
+        "REPHRASE": "Improve Writing",
+        "FIX_SPELLING_GRAMMAR": "Fix Spelling and Grammar",
+        "SHORTEN": "Shorten",
+        "EXPAND": "Expand",
+        "MAKE_FRIENDLY": "Make Friendly",
```
Comment (pranavrajs) [inline app/javascript/dashboard/i18n/locale/en/integrations.json:95]:
```suggestion
        "MAKE_FRIENDLY": "Change message tone to friendly",
```

File: app/javascript/dashboard/i18n/locale/en/integrations.json [hunk 7]
```diff
@@ -83,6 +83,28 @@
...
+        "MAKE_FORMAL": "Make Formal",
```
Comment (pranavrajs) [inline app/javascript/dashboard/i18n/locale/en/integrations.json:96]: C
```suggestion
        "MAKE_FORMAL": "Use formal tone",
```

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Looks nice, minor comments on the text. I also found that the rendering is not correct in some case. 
Please see the example below. 
Screenshot 2023-07-14 at 7 13 11 PM

**Comment by muhsin-k** [comment]
> Looks nice, minor comments on the text. I also found that the rendering is not correct in some case.

@pranavrajs Done.

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**